src/idl_gen_cpp.cpp

```diff
@@ -1706,6 +1708,158 @@ class CppGenerator : public BaseGenerator {
     code_ += "  }";
   }
 
+bool NeedsCopyCtorAssignOp(const StructDef &struct_def) {
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      const auto &field = **it;
+      const auto &type = field.value.type;
+      if (field.deprecated || type.base_type == BASE_TYPE_UTYPE) continue;
```

Comment:
why this checking for UTYPE in specific? it is just another scalar, so won't match the checks below anyway

Reply:
Ah, this code was refactored late into the patch. You're right that the BASE_TYPE_UTYPE check is not necessary here.

Reply:
Done.